src/main/java/org/terasology/launcher/platform/Platform.java

```diff
@@ -79,4 +96,22 @@ public String toString() {
     public static Platform getPlatform() {
         return PLATFORM;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Platform platform = (Platform) o;
+
+        if (os != platform.os) return false;
+        return arch == platform.arch;
+    }
```

Comment:
Previously, we only checked whether we are on a specific platform and had no means to create new instances of this class. Now that we want to manage the JRE for multiple platforms we have to open this up a bit.